File: SocialMediaAnalyticsDashboard/src/main/java/entity/Analytics.java
package entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Analytics {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private int followers;
    private int following;
    private int posts;
    private int accountReached;
    private int accountEngaged;
    private int contentShared;
    private int adsRun;
    private int activePromotions;
    private int totalStories;
    private int totalFollows;
    private int totalSaves;
    private int totalComments;
    private int totalShares;
}

File: social-media-analytics_frontend/src/components/Analytics.js
import React from 'react';

const Analytics = ({ analytics }) => (
    <div>
        <p>Followers: {analytics.followers}</p>
        <p>Following: {analytics.following}</p>
        <p>Posts: {analytics.posts}</p>
        <p>Account Reached: {analytics.accountReached}</p>
        <p>Account Engaged: {analytics.accountEngaged}</p>
        <p>Content Shared: {analytics.contentShared}</p>
        <p>Ads Run: {analytics.adsRun}</p>
        <p>Active Promotions: {analytics.activePromotions}</p>
        <p>Total Stories: {analytics.totalStories}</p>
        <p>Total Follows: {analytics.totalFollows}</p>
        <p>Total Saves: {analytics.totalSaves}</p>
        <p>Total Comments: {analytics.totalComments}</p>
        <p>Total Shares: {analytics.totalShares}</p>
    </div>
);

export default Analytics;
